app/uk/gov/hmrc/agentsubscription/NameManglingFilter.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscription
+import javax.inject._
+import play.api.GlobalSettings
+import play.api.http._
+import play.api.mvc.RequestHeader
+
+class NameManglingFilter @Inject() (provider: Provider[GlobalSettings])
+      extends GlobalSettingsHttpRequestHandler(provider) {
+
+  override def handlerForRequest(request: RequestHeader) = {
+    if (request.uri.startsWith("/agencies") || request.uri.startsWith("/sandbox")) {
```

Comment:
Javadoc for `request.uri`:
```
    /**
     * The complete request URI, containing both path and query string.
     * The URI is what was on the status line after the request method.
     * E.g. in "GET /foo/bar?q=s HTTP/1.1" the URI should be /foo/bar?q=s.
     * It could be absolute, some clients send absolute URLs, especially proxies.
     */
    def uri: String
```

I read `It could be absolute, some clients send absolute URLs, especially proxies` as saying that it might be fully absolute including a scheme and hostname. If so I think it's safer to use `request.path` instead (which also seems to work i.e. the tests pass).